lib/tty/pager/system.rb

```diff
@@ -158,6 +227,41 @@ def pager_command(*commands)
                            self.class.find_executable(*commands)
                          end
       end
+
+      # A wrapper for an external process.
+      #
+      # @api private
+      class PagerIO
+        def initialize(command)
+          @io  = IO.popen(command, 'w')
+          @pid = @io.pid
+        end
+
+        def write(*args)
+          io_call(:write, *args)
+        end
+
+        def puts(*args)
+          io_call(:puts, *args)
+        end
+
+        def close
+          @io.close
```

Comment:
Testing it out, calling `close` twice doesn't seem to error out, but I'm not sure if this is cross-platform behaviour, so I'll add a check.

It would be nice to write a test for it, but there wouldn't be a point if it uses stubs. I feel like this project could benefit a lot from integration tests with real process-spawning, but that's definitely not in the scope of this PR.

Reply:
💯 agree on integration tests and possibly performance tests as well with [rspec-benchmark](https://github.com/piotrmurach/rspec-benchmark) . I wrote things quite a while back and definitely the test suite needs more love.

Reply:
I've done integration tests for process-juggling before, and I was reasonably happy with how it turned out: [waiting_on_rails](https://github.com/AndrewRadev/waiting-on-rails/blob/master/spec/waiting_on_rails/rails_spec.rb). Essentially, the main idea was stubbing the `PATH` to ensure a fake `rails` command and a fake `mplayer` command get called, and those would be driven by the test suite via sockets. This is probably quite non-portable, though. Still, there are options.